packages/sdk/src/services/template-service.ts

```diff
@@ -0,0 +1,86 @@
+import { ItemTemplateApi, ItemTemplatesResponse } from '@meeco/vault-api-sdk';
+import { SDKTemplate } from '../models/sdk-template';
+import { getAllPaged, pagedToGenerator, reducePages } from '../util/paged';
+import Service, { IPageOptions, IVaultToken } from './service';
+
+/**
+ * List and fetch available templates for Meeco Items from the API.
+ */
+export class TemplateService extends Service<ItemTemplateApi> {
+  public getAPI(token: IVaultToken): ItemTemplateApi {
+    return this.vaultAPIFactory(token.vault_access_token).ItemTemplateApi;
+  }
+
+  public async list(
+    credentials: IVaultToken,
+    spec?: {
+      classificationScheme?: string;
+      classificationName?: string;
+      like?: string;
+    },
+    options?: IPageOptions
+  ): Promise<ItemTemplatesResponse> {
+    return this.vaultAPIFactory(credentials.vault_access_token).ItemTemplateApi.itemTemplatesGet(
+      spec?.classificationScheme,
+      spec?.classificationName,
+      spec?.like,
+      options?.nextPageAfter,
+      options?.perPage
+    );
+  }
+
+  public async listAll(
+    credentials: IVaultToken,
+    spec?: {
+      classificationScheme?: string;
+      classificationName?: string;
+      like?: string;
+    }
+  ): Promise<ItemTemplatesResponse> {
+    const api = this.vaultAPIFactory(credentials.vault_access_token).ItemTemplateApi;
+
+    return getAllPaged(cursor =>
+      api.itemTemplatesGet(spec?.classificationScheme, spec?.classificationName, spec?.like, cursor)
+    ).then(reducePages);
+  }
+
+  public async get(credentials: IVaultToken, id: string): Promise<SDKTemplate> {
+    const response = await this.vaultAPIFactory(
+      credentials.vault_access_token
+    ).ItemTemplateApi.itemTemplatesIdGet(id);
+    return SDKTemplate.fromAPI(response);
+  }
+
+  /**
+   * Search for an ItemTemplate by name, returning `undefined` if it is not found.
+   *
+   * Note that this patches a missing API method, so may take longer than expected to execute.
+   * Any call will cause at least two API calls, possibly many more if there are lots of templates.
+   */
+  public async findByName(
```

Comment:
Only real benefit here is that if it happens to be in the first few pages you make less requests?
Intuitively I feel like you're more than likely just going to fetch the full list and find by name in memory.

Reply:
For a web app ok, but not for CLI right? I had hoped this one would be replaced by a new API filter param on /templates/get eventually.

Reply:
Fair point